qtodotxt/ui/controllers/tasks_list_controller.py

```diff
@@ -105,6 +107,11 @@ def _initIncreasePrioritySelectedTasksAction(self):
 
     def completeTask(self, task):
         if not task.is_complete:
+            # Check if task is recurrent
+            # TODO: Many variables checked here - does it work smarter, too?
+            if task.recIncr is not None and task.recInterv is not None \
+                    and task.recMode is not None and task.due is not None:
+                self._recurTask(task)
```

Comment:
Yes, but I can make this test also simpler if I just would check the variable `task.recMode is not None`
I've redesigned the class structure(s) to your suggestions. Thank you.